### PR TITLE
feat(core): `verifyAndParseCodeFromCallbackUri`

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(core): `verifyAndParseCodeFromCallbackUri`

## Testing

UT passed.
image

### CODE REVIEW

File: core/verify_and_parse_code_from_callback_uri.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package core
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func VerifyAndParseCodeFromCallbackUri(callbackUri, redirectUri, state string) (string, error) {
```
Comment (simeng-li) [inline core/verify_and_parse_code_from_callback_uri.go:10]: Do callbackUri and redirectUri need a type assertion?  Will strings.HasPrefix throw if callbackUri or redirectUri is not a string?

Comment (xiaoyijun) [reply core/verify_and_parse_code_from_callback_uri.go:10]: in Golang:
```go
func VerifyAndParseCodeFromCallbackUri(callbackUri, redirectUri, state string) 
```
is equal to 
```go
func VerifyAndParseCodeFromCallbackUri(callbackUri string, redirectUri string, state string) 
```

**Review by charIeszhao — APPROVED** [review]
LGTM. Better wait for a second approver, though. Cheers!

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**